enhancements/olm/olm-managed-operator-metrics.md

```diff
@@ -0,0 +1,145 @@
+
+#### ServiceMonitor and PrometheusRule Support
+
+As OLM gains support for these Prometheus objects, there is a chance that installed operators may overload the Prometheus instance. It is important that a Cluster Admin is alerted that an operator will report metrics prior to deploying the operator. As such, we propose adding a `monitoringSupported` boolean field to the `ClusterServiceVersion` object. OLM will not install either of the Prometheus object unless the `monitoringSupported` field is set to true. Additionally, Cluster Admins will need to manually the operator's `InstallPlans` in the following scenarios:
+
+- A fresh install for this operator is being created where this field is set to `true`
+- An existing operator had this field set to `false` and is being upgraded to a new version that sets this field to `true`
+
+> Note: The `InstallPlan` will include a helpful message for Cluster Admins if either of the situations above are encountered.
+
+### OLM Telemeter Support
+
+#### Telemeter Requirements
+
+Once OLM is capable of managing `ServiceMonitor` and `PrometheusRule` objects along with the lifecycle of an operator, OLM must allow Red Hat owned operators to take advantage of Telemeter. Before moving forward, let's review the requirements that must be met for Telemeter to scrape an application's metrics:
+
+1. The application must be deployed in a namespace that is both prefixed with `openshift-` and labeled with `openshift.io/cluster-monitoring=true`
+2. A [ServiceMonitor object](https://github.com/coreos/prometheus-operator/blob/master/Documentation/user-guides/getting-started.md#related-resources) must be created in the namespace mentioned above and point to the application's metrics endpoint
+3. A role/rolebinding must be created that provides the Prometheus Operator ServiceAccount with the appropriate RBAC priviledges to discover the `ServiceMonitors` and `PrometheusRules` objects in the newly created namespace.
+
+Both requirements 1 and 2 are fully met by the changes proposed above leaving our team with the duty to fulfill requirement 3.
+
+#### Fulfilling Requirement 3: Updating the Prometheus Operator ServiceAccount
+
+In order for the Prometheus Operator to recognize take advantage of this new feature, its ServiceAccount must have the proper RBAC permissions to view events on the `ServiceMonitors` and `PrometheusRules` objects in the operator's namespace.
+
+On upstream clusters, there is no way for OLM to identify which Prometheus Operator ServiceAccount should be granted the RBAC priviledges. As such, the cluster admin will need to update the correct Prometheus Operator's OperatorGroup to include the correct namespace.
+
+On OpenShift cluster, OLM can simply update the default Prometheus instance with the correct RBAC priviledges when the `openshift.io/cluster-monitoring=true` label is present on the namespace.
```

Comment:
I would think that this needs to be a combination of the UI and the OLM to complete.

In the UI Stories, we talked about granting the cluster-admin the ability to choose if they enable this, how does the UI signal "don't actually add this operator to monitoring?"

Reply:
@ecordell and I had discussed this offline prior to submitting the proposal.

For the first round implementation , I suggest only allowing the operator to install if the Cluster Admin agrees that it should provide metrics to monitoring.

In a future sprint, I suggest that OLM provide a `Subscription Override` feature that manages whether or not monitoring is enabled for the operator.